src/pluto_eris/curve.rs

```diff
@@ -0,0 +1,283 @@
+        0x42fdbb92e64ba85a,
+        0x58e88d4df1e7418c,
+        0x2239f7408ead478c,
+    ]),
+
+    // 0x1260b04d51136590dbb53dfd7caf450aeca714555bbe4f079ca65d97eb28fc9fc697b4e10bbcd9e0539ef82a731fb88ed49e3c080e6d945d
+    c1: Fp::from_raw([
+        0xd49e3c080e6d945d,
+        0x539ef82a731fb88e,
+        0xc697b4e10bbcd9e0,
+        0x9ca65d97eb28fc9f,
+        0xeca714555bbe4f07,
+        0xdbb53dfd7caf450a,
+        0x1260b04d51136590,
+    ]),
+};
+
+const TRITON_A: Fp2 = Fp2::ZERO;
+
+// u + 3
+const TRITON_B: Fp2 = Fp2 {
+    c0: Fp::from_raw([0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00]),
+    c1: Fp::ONE,
+};
+
+impl group::cofactor::CofactorGroup for G1 {
+    type Subgroup = G1;
+
+    fn clear_cofactor(&self) -> Self {
+        *self
+    }
+
+    fn into_subgroup(self) -> CtOption<Self::Subgroup> {
+        CtOption::new(self, 1.into())
+    }
+
+    fn is_torsion_free(&self) -> Choice {
+        1.into()
+    }
+}
+
+new_curve_impl!(
+    (pub),
+    G1,
+    G1Affine,
+    false,
+    Fp,
+    Fq,
+    (G1_GENERATOR_X,G1_GENERATOR_Y),
+    PLUTO_A,
+    PLUTO_B,
+    "pluto",
+    |curve_id, domain_prefix| svdw_hash_to_curve(curve_id, domain_prefix, G1::SVDW_Z),
+);
+
+impl group::cofactor::CofactorGroup for Eris {
+    type Subgroup = Eris;
+
+    fn clear_cofactor(&self) -> Self {
+        *self
+    }
+
+    fn into_subgroup(self) -> CtOption<Self::Subgroup> {
+        CtOption::new(self, 1.into())
+    }
+
+    fn is_torsion_free(&self) -> Choice {
+        1.into()
+    }
+}
+
+impl G1 {
+    const SVDW_Z: Fp = Fp::ONE;
+}
+
+new_curve_impl!(
+    (pub),
+    Eris,
+    ErisAffine,
+    false,
+    Fq,
+    Fp,
+    (ERIS_GENERATOR_X,ERIS_GENERATOR_Y),
+    ERIS_A,
+    ERIS_B,
+    "eris",
+    |curve_id, domain_prefix| svdw_hash_to_curve(curve_id, domain_prefix, Eris::SVDW_Z),
+);
+
+impl CofactorGroup for G2 {
+    type Subgroup = G2;
+
+    fn clear_cofactor(&self) -> Self {
+        // cofactor = 2*p - q
+        //0x24000000000024000130e0000d7f70e4a803ca76f439266f443f9a5d3a8a6c7be4a7d5fe91447fd6a8a7e928a00867971ffffcd300000001
+        let e: [u8; 56] = [
+            0x24, 0x00, 0x00, 0x00, 0x00, 0x00, 0x24, 0x00, 0x01, 0x30, 0xe0, 0x00, 0x0d, 0x7f,
+            0x70, 0xe4, 0xa8, 0x03, 0xca, 0x76, 0xf4, 0x39, 0x26, 0x6f, 0x44, 0x3f, 0x9a, 0x5d,
+            0x3a, 0x8a, 0x6c, 0x7b, 0xe4, 0xa7, 0xd5, 0xfe, 0x91, 0x44, 0x7f, 0xd6, 0xa8, 0xa7,
+            0xe9, 0x28, 0xa0, 0x08, 0x67, 0x97, 0x1f, 0xff, 0xfc, 0xd3, 0x00, 0x00, 0x00, 0x01,
+        ];
+
+        // self * TRITON_COFACTOR
+        let mut acc = G2::identity();
+        for bit in e
+            .iter()
+            .flat_map(|byte| (0..8).rev().map(move |i| Choice::from((byte >> i) & 1u8)))
+            .skip(1)
+        {
+            acc = acc.double();
+            acc = G2::conditional_select(&acc, &(acc + self), bit);
+        }
+        acc
+    }
+
+    fn into_subgroup(self) -> CtOption<Self::Subgroup> {
+        unimplemented!()
```

Comment:
This is inherited from the `bn246` implementation. I believe the reason for having this unimplemented is that we are not using the trait as it is intended.
We are implementing `CofactorGroup` for `G2`.  The associated type for `Subgroup` should not be `G2` but a different struct. 
In relation to that, the result of `clear_cofactor` should be `Self::Subgroup` not `Self`, it just happens that this is ok for most curves, since they have cofactor 1.

I will open an issue detailing this problem and other changes that also affect the implementation of `bn256`. In this way we can address them there as well.